### PR TITLE
6 get api articles comments

### PR DESCRIPTION
**Pull request by swamphobbit22**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -109,3 +129,47 @@ it('should return an array of articles' , () => {
...
+      body.comments.forEach((comment) => {
+        expect(comment).toMatchObject({
+          comment_id: expect.any(Number),
+          votes: expect.any(Number), 
+          created_at: expect.any(String) ,
+          author: expect.any(String) ,
+          body: expect.any(String),
+          article_id: expect.any(Number),
+        })
+      })
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:148]: I would also check the length of the array here and make sure that the article_id is 3!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -109,3 +129,47 @@ it('should return an array of articles' , () => {
...
+ it('should respond with 404 error with an id that does not exist', () => {
+  return request(app)
+  .get('/api/articles/999/comments')
+  .expect(404)
+  .then(({ body }) => {
+    const { msg } = body;
+    expect(msg).toBe('article not found')
+  })
+})
+
```
Comment (nc-christian-dutton) [inline __tests__/app.test.js:173]: Good error testing! 

there's another test you can add - if an article exists but has no comments yet we should get a 200 and an empty array. you will have to check that the article exists!

File: db/test.sql [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+        SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url,
+        COUNT(comments.comment_id) AS comment_count
+        FROM articles
+        LEFT JOIN comments
+        ON comments.article_id = articles.article_id
+        GROUP BY articles.article_id
+        ORDER BY articles.created_at DESC;
+
+        SELECT * FROM comments;
```
Comment (nc-christian-dutton) [inline db/test.sql:18]: I would gitignore this file :)

File: errors/error-handling.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+exports.customErrorhandler = (err, req, res, next) => {
+    if (err.status) {
+        res.status(err.status).send({ msg: err.msg });
+      } else next(err);
+};
+
+exports.serverErrorhandler = (err, req, res, next) => {
+    console.log(err);
+    res.status(500).send({ msg: "Internal Server Error" });
+};
```
Comment (nc-christian-dutton) [inline errors/error-handling.js:16]: Good job extracting these out!
